pkg/domain/entities/params/generator/generator.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"text/template"
+)
+
+var bodyTmpl = `package {{.PackageName}}
+
+import (
+{{range $import := .Imports}}	{{$import}}
+{{end}}
+)
+`
+
+var fieldTmpl = `
+// With{{.Name}}
+func(o *{{.StructName}}) With{{.Name}}(value {{.Type}}) *{{.StructName}} {
+	v := &value
+	o.{{.Name}} = v
+	return o
+}
+`
+
+var changed = `
+// Changed
+func (o *{{.StructName}}) Changed(fieldName string) bool {
+	r := reflect.ValueOf(o)
+	value := reflect.Indirect(r).FieldByName(fieldName)
+	return !value.IsNil()
+}
+`
+
+func writeChangedFunc(f *os.File, structName string) error {
+	changedStruct := struct {
+		StructName string
+	}{
+		StructName: structName,
+	}
+
+	t, err := template.New(structName).Parse(changed)
```

Comment:
You should move this parsing out of function and use MustParse(). No reason to build and keep a template per field.